src/providers/google/youtube-following.ts

```diff
@@ -0,0 +1,170 @@
+            syncPosition = this.stopSync(syncPosition);
+
+            return {
+                position: syncPosition,
+                results: [],
+            };
+        }
+
+        const results = await this.buildResults(
+            youtube,
+            serverResponse,
+            syncPosition.breakId,
+            _.has(this.config, "metadata.breakTimestamp")
+                ? this.config.metadata.breakTimestamp
+                : undefined
+        );
+
+        syncPosition = this.setNextPosition(syncPosition, serverResponse);
+
+        if (results.length != this.config.batchSize) {
+            // Not a full page of results, so stop sync
+            syncPosition = this.stopSync(syncPosition);
+        }
+
+        return {
+            results,
+            position: syncPosition,
+        };
+    }
+
+    protected stopSync(syncPosition: SyncHandlerPosition): SyncHandlerPosition {
+        if (syncPosition.status == SyncHandlerStatus.STOPPED) {
+            return syncPosition;
+        }
+
+        syncPosition.status = SyncHandlerStatus.STOPPED;
+        syncPosition.thisRef = undefined;
+        syncPosition.breakId = syncPosition.futureBreakId;
+        syncPosition.futureBreakId = undefined;
+
+        return syncPosition;
+    }
+
+    protected setNextPosition(
+        syncPosition: SyncHandlerPosition,
+        serverResponse: GaxiosResponse<youtube_v3.Schema$SubscriptionListResponse>
+    ): SyncHandlerPosition {
+        if (!syncPosition.futureBreakId && serverResponse.data.items.length) {
+            syncPosition.futureBreakId = `${this.connection.profile.id}-${serverResponse.data.items[0].id}`;
+        }
+
+        if (_.has(serverResponse, "data.nextPageToken")) {
+            // Have more results, so set the next page ready for the next request
+            syncPosition.thisRef = serverResponse.data.nextPageToken;
+        } else {
+            syncPosition = this.stopSync(syncPosition);
+        }
+
+        return syncPosition;
+    }
+
+    protected async buildResults(
+        youtube: youtube_v3.Youtube,
+        serverResponse: GaxiosResponse<youtube_v3.Schema$SubscriptionListResponse>,
+        breakId: string,
+        breakTimestamp?: string
+    ): Promise<SchemaFollowing[]> {
+        const results: SchemaFollowing[] = [];
+        for (const item of serverResponse.data.items) {
+            const itemId = `${this.connection.profile.id}-${item.id}`;
+
+            if (itemId == breakId) {
+                break;
+            }
+
+            const snippet = item.snippet;
+            const contentDetails = item.contentDetails;
+            const insertedAt = snippet.publishedAt || "Unknown";
+
+            if (breakTimestamp && insertedAt < breakTimestamp) {
+                break;
+            }
+
+            const title = snippet.title || "No title";
+
+            results.push({
+                _id: `youtube-following-${itemId}`,
+                name: title,
```

Comment:
Other fields that should be added if available; icon, uri.

Also populate followedTimestamp and sourceData (see facebook following)

Reply:
Can you provide an example of all the data available in `item`? I'd like to see if there's more data we can include.

Reply:
Please check answer below